src/hooks/usePriceEstimation.ts

```diff
@@ -11,13 +11,24 @@ interface Params {
   quoteTokenId: number
 }
 
-export function usePriceEstimation(params: Params): BigNumber | null {
+interface Result {
+  priceEstimation: BigNumber | null
+  isPriceLoading: boolean
+}
+
+export function usePriceEstimation(params: Params): Result {
   const { baseTokenId, quoteTokenId } = params
-  const [price, setPrice] = useSafeState<BigNumber | null>(null)
+  const [isPriceLoading, setIsPriceLoading] = useSafeState(true)
+  const [priceEstimation, setPriceEstimation] = useSafeState<BigNumber | null>(null)
 
   useEffect(() => {
-    getPriceEstimation({ baseTokenId, quoteTokenId }).then(setPrice, logDebug)
-  }, [quoteTokenId, baseTokenId, setPrice])
+    setIsPriceLoading(true)
+
+    getPriceEstimation({ baseTokenId, quoteTokenId }).then(price => {
+      setPriceEstimation(price)
+      setIsPriceLoading(false)
```

Comment:
No catch? you don't need to do a `setIsPriceLoading(false)` also if you fail. Also handle the error somehow?

Reply:
Check the next line https://github.com/gnosis/dex-react/pull/762/files#diff-c47f9093788861eb688116377dd46937R30

But I get your point.
Will address

Reply:
Done undoing @Velenir 's improvement suggestion